apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue

```diff
@@ -5,69 +5,57 @@
-        v-text="hours.timeString" />
+        v-text="hours.timeString === CLOSED ? translate(CLOSED) : hours.timeString" />
     </div>
-
-    <DoButton
-      v-if="!resolvedShowAll"
-      link
-      @click="mutableShowAll = true">
-      {{ translate(SHOW_MORE_HOURS) }}
-    </DoButton>
   </div>
 </template>
 
 <script lang="ts" setup>
-import {capitalize, computed, onDeactivated, ref, toRefs} from 'vue';
-import {addDays, isSameDay, isToday} from 'date-fns';
-import {CLOSED, OPENING_HOURS, SHOW_MORE_HOURS} from '@myparcel-do/shared';
-import {type StartEndDate} from '@myparcel/sdk';
-import {createNextDate, createUtcDate} from '../../../../utils';
-import {SHOWN_OPENING_HOURS} from '../../../../data';
-import {useDateFormat, useLanguage, usePickupLocation, useTimeRange} from '../../../../composables';
-import {DoButton} from '../../../../components';
+import {computed, toRef} from 'vue';
+import {OPENING_HOURS, CLOSED} from '@myparcel-do/shared';
+import {useLanguage, usePickupLocation} from '../../../../composables';
 
 const props = defineProps<{locationCode: string; expanded?: boolean}>();
-const propRefs = toRefs(props);
 
 const {translate} = useLanguage();
+const locationCodeRef = toRef(props, 'locationCode');
+const {pickupLocation} = usePickupLocation(locationCodeRef);
 
-const {pickupLocation} = usePickupLocation(propRefs.locationCode);
-
-const mutableShowAll = ref(false);
-const resolvedShowAll = computed(() => props.expanded || mutableShowAll.value);
+const weekDaysMap = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
 
 const openingHours = computed(() => {
-  return (pickupLocation.value?.openingHours ?? [])
-    .map(({hours, weekday}) => {
-      const date = createNextDate(weekday);
-      const formattedDay = useDateFormat(date);
-
-      const isTodayOrTomorrow = isToday(date) || isSameDay(addDays(createUtcDate(), 1), date);
-
-      const time: StartEndDate = hours?.[0];
-      const timeString = time ? useTimeRange(time.start.date, time.end.date).value : translate(CLOSED);
-
-      return {
-        date,
-        weekday: capitalize(isTodayOrTomorrow ? formattedDay.relative.value : formattedDay.weekday.value),
-        timeString,
-      };
-    })
-    .sort((a, b) => a.date.getTime() - b.date.getTime());
-});
-
-const filteredOpeningHours = computed(() => {
-  return resolvedShowAll.value ? openingHours.value : openingHours.value.slice(0, SHOWN_OPENING_HOURS);
-});
-
-onDeactivated(() => {
-  mutableShowAll.value = false;
+  const openingHoursData = pickupLocation.value?.openingHours ?? [];
+
+  return weekDaysMap.map((day, index) => {
+    const dayData = openingHoursData.find((entry: {weekday: number}) => entry.weekday === index);
+
+    if (dayData?.hours && dayData.hours.length > 0) {
+      const times = dayData.hours[0];
+
+      if (times.start && times.end) {
+        const timeString = `${times.start.date.substring(11, 16)} - ${times.end.date.substring(11, 16)}`;
```

Comment:
I don't like having random numbers in code, as its hard to see what they signify. Prefer wrapping these in named constants so it's immediately clear what they're intending to do. This code currently seems pretty error-prone, but it's hard to say without knowing how consistent the `date` input string is supposed to be -- prefer to work with dates using `Date` or `date-fns` rather than string data.